pkg/controller/nodegroup/nodegroup_controller.go

```diff
@@ -218,7 +222,27 @@ func (r *ReconcileNodeGroup) Reconcile(request reconcile.Request) (reconcile.Res
 		return reconcile.Result{}, err
 	}
 
-	// TODO: Add parameters to obj, and update if parameters change.
+	cfnParameters, err := getCFNParametersFromCFNStack(cfnSvc, instance)
+	if err != nil {
+		r.fail(instance, "error trying to read the CFN Parameters", err, logger)
+		return reconcile.Result{}, err
+	}
+
+	if shouldUpdate(crdParameters, cfnParameters) {
```

Comment:
So if User make changes to CRD for Version or IAMPolicies, the update will not be invoked. It means user has to find workaround to invoke update for  Version or IAMPolicies by updating any instance parameter. 

Can we include Version and IAMPolicies changes inclusive in shouldUpdate?

Reply:
Good call out on the Version, we have already faced a scenario where this is needed- I will submit this as a new PR as changes need to be applied for both the controlPlane and NodeGroup. 
As for the IAM policies the current way we check for updates is by reading the Parameters of the CFN Template, the IAM policies cannot be moved to the Parameters section, hence we will need to parse the CFN template body to get this value. I would defer from going down this route for now.

Reply:
Please create an issue to address this one.

Reply:
https://github.com/awslabs/aws-eks-cluster-controller/issues/85

Reply:
#86